Lib/logging/handlers.py

```diff
@@ -329,6 +329,10 @@ def computeRollover(self, currentTime):
                     else:
                         daysToWait = 6 - day + self.dayOfWeek + 1
                     result += daysToWait * (60 * 60 * 24)
+            if self.when == 'MIDNIGHT':
```

Comment:
This could be an `elif`, right?

Reply:
No, because the `else` branch should be executed when `self.when.startswith('W')`.

I see now that two `if`s can be merged:

```python
            if self.when.startswith('W'):
                day = currentDay # 0 is Monday
                if day != self.dayOfWeek:
                    if day < self.dayOfWeek:
                        daysToWait = self.dayOfWeek - day
                    else:
                        daysToWait = 6 - day + self.dayOfWeek + 1
                    result += daysToWait * _MIDNIGHT
                result += self.interval - _MIDNIGHT * 7
            else:
                result += self.interval - _MIDNIGHT